src/lexer/regexFactory.ts

```diff
@@ -76,6 +76,33 @@ export const parameter = (paramTypes: string[], pattern: string): RegExp | undef
   return patternToRegex(`(?:${typesRegex})(?:${pattern})`);
 };
 
+const buildQStringPatterns = () => {
+  const specialDelimiterMap = {
+    '<': '>',
+    '[': ']',
+    '(': ')',
+    '{': '}',
+  };
+
+  // base pattern for special delimiters, left must correspond with right
+  const singlePattern = "{left}(?:[^{right}]|{right}(?!'))+{right}";
+
+  // replace {left} and {right} with delimiters, collect as array
+  const patternList = Object.entries(specialDelimiterMap).map(([left, right]) =>
+    singlePattern.replace(/{left}/g, escapeRegExp(left)).replace(/{right}/g, escapeRegExp(right))
+  );
+
+  // standard pattern for common delimiters, ignores keys of specialDelimiterMap
+  const standardDelimiterPattern = String.raw`(?<tag>[^\s${escapeRegExp(
+    Object.keys(specialDelimiterMap).join('')
+  )}])(?:(?!\k<tag>).|\k<tag>(?!'))+\k<tag>`;
```

Comment:
I'm not sure whether this regex is correct. There are lots of negative lookaheads :)

Specifically It's this middle part that concerns me:

```
(?:(?!\k<tag>).|\k<tag>(?!'))+
```

Wouldn't we get the same result with:

```
(?:(?!\k<tag>').)+
```

Didn't run the code. Pretty tired today as well :)

Reply:
I also notice that we are using a greedy quantifier `+` here. The regex might end up simpler if we instead used a non-greedy `+?`.